src/cursor.js

```diff
@@ -10,8 +10,21 @@
 import {valToKeyPath, makeCursor} from './utils';
 import createAtom from 'atom-store';
 
+const inMemory = (initial, transition) => {
+  let rootState = initial;
+  const read = () => rootState;
+  const write = (fn, ...context) => {
+    const oldState = read();
+    const newState = fn(oldState);
+    transition(newState, oldState, ...context);
```

Comment:
I'm in two minds as to whether providing the ability to add arbitrary arguments from `write()` through to the `onChange` callback(s), should be the responsibility of the consumer application, or in fact, should be an implementation concern for the core `atom-store` plugin itself. My hunch, is that it _could_ be a relatively common requirement from consumers other than just `immutable-cursor`.

What are your thoughts?